gtests/net/mptcp/mp_join/mp_join_server_v4.pkt

```diff
@@ -17,22 +18,18 @@
 +0    accept(3, ..., ...) = 4
 
 // add_addr
-+0.0    >                                .  1:1(0)  ack 1             <nop, nop,         TS val 4074418293 ecr 4074418292, add_address addr[saddr1] hmac=auto, dss dack4=1 dll=0 nocs>
-// TODO: send echo bit
-// +0.0 <                                .  1:1(0)  ack 1             <nop, nop,         TS val 4074418293 ecr 4074418292, add_address addr[saddr1],           dss dack8=1 dll=0 nocs>
-
-
-+0.2    <                               P.  1:3(2)  ack 1  win 256    <nop, nop,         TS val 4074418292 ecr 4074410674,                mpcapable v1 flags[flag_h] key[skey, ckey] mpcdatalen 2, nop, nop>
-// MP_CAPABLE carrying data are acked with 64-bit, safer not knowing if the
-// sender will use a DSN on 64-bit or 32-bit. Later we can use the lower 32 bits
-+0.0    >                                .  1:1(0)  ack 3             <nop, nop,         TS val 4074418293 ecr 4074418292,                dss dack8=3 dll=0 nocs>
```

Comment:
out of curiosity, why removing these last 4 lines? Should we not keep them?

Reply:
@matttbe because we have a bidirectional exchange of DSS on lines 21 and 22. I'm not sure if it's legal to send MP_CAPABLE with data after that (but indeed, I didn' even check if the kernel under test accepts these 2 bytes)

Reply:
I didn't find anything in the RFC not allowing that but I maybe read this too fast.

For me, it is OK because this 4th MP_CAPABLE ACK is there to make sure we don't have middleboxes or server not handling MPTCP but sending "dummy" MPTCP options. So I think the sender can start sending ADD_ADDR (and receive the echo one) before the receiver sends its first data packet with the 4th MP_CAPABLE ACK. I don't think it is mandatory for the receiver to send a dedicated 4th MP_CAPABLE ACK then the echo ADD_ADDR.

@cpaasch sorry to bother you on this one but according to the MPTCPv1 specs, can we exchange (bidirectionally) ADD_ADDR before the 4th MP_CAPABLE ACK?

    >  S   MPC
    <  S.  MPC + key B
    >   .  MPC + key A,B

    >   .  ADD_ADDR
    <   .  ADD_ADDR echo

    <   P.  MPC + key A,B + data

Reply:
> For me, it is OK because this 4th MP_CAPABLE ACK is there to make sure we don't have middleboxes or server not handling MPTCP but sending "dummy" MPTCP options. So I think the sender can start sending ADD_ADDR (and receive the echo one) before the receiver sends its first data packet with the 4th MP_CAPABLE ACK. I don't think it is mandatory for the receiver to send a dedicated 4th MP_CAPABLE ACK then the echo ADD_ADDR.

IIRC, the purpose of the MPC + data here is to ensure that the next outbound DACK is 64-bit wide.  I will try it (and convert the inbound DACKs carried together to be 32-bit wide).